tests/testthat/test_query.R

```diff
@@ -74,73 +72,39 @@ test_that("returns count result on empty 1-column timeseries", {
   names(columns) <- c(column_name)
 
   handle <- connect(qdbd$uri)
-  ts_create(handle,
-                name = alias,
-                columns = columns)
+  ts_create(handle, name = alias, columns = columns)
+  ts_insert.double(handle, name = alias, column = column_name)
   results <-
-    query(handle,
-              sprintf("SELECT COUNT(*) FROM %s IN RANGE(2017, +1y)", alias))
+    query(handle, sprintf("SELECT COUNT(*) FROM %s IN RANGE(2017, +1y)", alias))
 
   expect(is.list(results), failure_message = "query result should be a list")
+  expect_named(results, c("scanned_point_count", "column_count", "columns", "row_count", "rows"))
 
-  expect_member(results, "scanned_point_count")
-  expect_equal(results$scanned_point_count, 0)
-
-  expect_member(results, "tables_count")
-  expect_equal(results$tables_count, 1)
-
-  # Check tables
-  expect_member(results, "tables")
-
-  tables <- results$tables
-  expect_equal(length(tables), 1)
-
-  expect_member(tables, alias)
-  expect(
-    alias %in% names(tables),
-    failure_message = sprintf(
-      "tables should contain <timeseries name>: [%s]",
-      paste(names(tables), collapse = ", ")
-    )
-  )
-
-  # Check table
-  table <- tables[[alias]]
-
-  expect_member(table, "columns_count")
-  expect_equal(table$columns_count, 2)
-
-  expect_member(table, "rows_count")
-  expect_equal(table$rows_count, 1)
+  expect_equal(results$scanned_point_count, 3)
+  expect_equal(results$column_count, 1)
 
   # Check columns
-  expect_member(table, "columns")
-
-  actual_columns <- table$columns
-  expect_equal(actual_columns,
-               c("$timestamp", sprintf("count(%s)", column_name)))
-
-  # Check data
-  expect_member(table, "data")
-
-  data <- table$data
-  expect(is.data.frame(data), failure_message = "data should be a data.frame")
-  expect_equal(colnames(data),
-               c("$timestamp", sprintf("count(%s)", column_name)))
-  expect_equal(rownames(data), c("1"))
-  expect_equal(dim(data), c(1, 2))
-
-  expect_na(data[["$timestamp"]])
-  expect_equal(data[[sprintf("count(%s)", column_name)]], 0)
+  count_result_column <- sprintf("count(%s)", column_name)
+
+  actual_columns <- results$columns
+  expect_equal(actual_columns, c(count_result_column))
+  
+  expect_equal(results$row_count, 1)
+  
+  rows <- results$rows
+  expect(is.data.frame(rows), failure_message = "rows should be a data.frame")
+  expect_equal(colnames(rows), c(count_result_column))
+  expect_equal(dim(rows), c(1, 1))
+  expect_equal(rows[[count_result_column]], 3)
 })
 
 test_that("returns count result on empty multi-column timeseries", {
   alias <- generate_alias("timeseries")
   columns <-
-    c(column_type$blob,
+    c(column_type$double,
```

Comment:
Why?

Reply:
Because I then insert data, sure I could add the insert function for other types, but the tests are not through enough to consider that in the first place.